File: problem-1/problem-1.rs
use std::env;

fn main() {
  let args: Vec<String> = env::args().collect();
  let val: i32 = args[0].parse().unwrap();
  println!("Answer: {}",euler_1(val));
}

fn euler_1(x: i32) -> i32 {
  let mut result: i32 = 0;
  let x = x/3;
  let y = x/5;
  let mut i = 1;

  while i < x {
    result += 3*i;
    i += 1;
  }
  i = 1;

  while i < y {
    if 5*i % 3 != 0 {
      result += 5*i;
      i += 1;
    }
  }

  result
}

File: problem-1/problem_1.php
<?php
function euler_1($n) {
    $result = [
        'terms' => [],
        'answer' => 0
    ];
    $x = $n/3;
    $y = $n/5;
    for ($i = 1;$i<$x;$i++) {
        $result['answer'] += 3 * $i;
        array_push($result['terms'],3 * $i);
    }
    for ($i = 1;$i<$y;$i++) {
        if (!(5*$i % 3 == 0)) {
            $result['answer'] += 5 * $i;
            array_push($result['terms'],5 * $i);
        }
    }
    sort($result['terms']);
    if (count($result['terms']) == 0) {
        $result['answer'] = 0;
    }
    return json_encode($result);
}

$v = $_REQUEST['val'];

echo euler_1($v);
?>
